### PR TITLE
Create `LocalStorageWallet` class to handle the wallet interaction

### LINKED ISSUE #4431: Create `LocalStorageWallet` class to handle the wallet interaction
### Description

We need to create `LocalStorageWallet` class which implements `BaseWallet` interface. It should have all the logic implemented in the following. 

https://github.com/ChainSafe/web3.js/blob/50afcd0318112b5da12f4944d3fdbdbb2f862ca2/packages/web3-eth-accounts/src/index.js#L617-L621

### Acceptance Criteria

- The class `load` and `save` behaviour works only when `localStorage` is available. Else it should throw error. 
- The class should be exported from `web3-core` package
- Any utility structure for that class should not be exported from the package. 

### Additional Information

- The class should use `scrypt-js` package for any encryption requirement

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Create a local storage wallet implementation. 

Fixes #4431

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-core/src/local_storage_wallet.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+				? storage
+				: undefined;
+		}
+	}
+
+	public get length() {
+		return Object.keys(this._accounts).length;
+	}
+
+	public create(numberOfAccounts: number, entropy: string) {
```
Comment (luu-alex) [inline packages/web3-core/src/local_storage_wallet.ts:56]: Looking at [1.x](https://github.com/ChainSafe/web3.js/blob/50afcd0318112b5da12f4944d3fdbdbb2f862ca2/packages/web3-eth-accounts/src/index.js#L129) if an entropy isn't provided should be generated with `randomHex`.

Was looking at the wrong function oops

File: packages/web3-core/src/local_storage_wallet.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	public remove(addressOrIndex: string | number): boolean {
+		if (typeof addressOrIndex === 'string') {
+			delete this._accounts[addressOrIndex];
+			return true;
+		}
+
+		const { address } = Object.values(this._accounts)[addressOrIndex];
+		delete this._accounts[address];
+
+		return true;
```
Comment (luu-alex) [inline packages/web3-core/src/local_storage_wallet.ts]: following [1.x](https://github.com/ChainSafe/web3.js/blob/50afcd0318112b5da12f4944d3fdbdbb2f862ca2/packages/web3-eth-accounts/src/index.js#L691) we should return false if the account does not exist

File: packages/web3-core/src/local_storage_wallet.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+> extends Web3BaseWallet<T> {
+	private readonly _accounts: { [key: string]: T };
+	private readonly _defaultKeyName = 'web3js_wallet';
+
+	public constructor(accountProvider: Web3AccountProvider<T>) {
+		super(accountProvider);
+		this._accounts = {};
+	}
+
+	public static getStorage(): Storage | undefined {
```
Comment (luu-alex) [inline packages/web3-core/src/local_storage_wallet.ts:23]: In [1.x](https://github.com/ChainSafe/web3.js/blob/50afcd0318112b5da12f4944d3fdbdbb2f862ca2/packages/web3-eth-accounts/src/index.js#L768) this function is named `storageAvailable`, perhaps we can make an alias so it's not breaking

**Comment by luu-alex** [comment]
change `local_stroage_wallet.test.ts` -> `local_storage_wallet.test.ts`, otherwise lgtm

File: packages/web3-core/src/local_storage_wallet.ts (hunk 3 above)
Comment (nazarhussain) [reply packages/web3-core/src/local_storage_wallet.ts:23]: Its a private function and never exposed outside. So its not a breaking change.

**Review by luu-alex — APPROVED** [review]

File: packages/web3-common/src/web3_base_wallet.ts [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
+	privateKeyToAccount: (privateKey: string) => T;
+	create: (entropy: string) => T;
+	decrypt: (keystore: string, password: string, options?: Record<string, unknown>) => T;
+}
+
+export abstract class Web3BaseWallet<T extends Web3BaseWalletAccount> {
+	protected readonly _accountProvider: Web3AccountProvider<T>;
+
+	public constructor(accountProvider: Web3AccountProvider<T>) {
+		// eslint-disable-next-line no-underscore-dangle
```
Comment (Spacesai1or) [inline packages/web3-common/src/web3_base_wallet.ts:22]: I prefer the use of `_` for protected/private properties in classes. To refrain from having one-offs where we disable the rule inline, can we just disable the rule for the whole project?

Comment (Spacesai1or) [reply packages/web3-common/src/web3_base_wallet.ts:22]: Created #4461 to track this

**Review by Spacesai1or — APPROVED** [review]
#4461 still needs to be addressed, but approving this so Nazar can move forward since we're 12hrs apart

**Pull request merged**